File: android/src/main/java/com/vivekjm/fastvideo/FastVideoRecords.kt
package com.vivekjm.fastvideo

import expo.modules.kotlin.records.Field
import expo.modules.kotlin.records.Record

class FastVideoDrm : Record {
  @Field var type: String = ""
  @Field var licenseUrl: String = ""
  @Field var certificateUrl: String? = null
  @Field var headers: Map<String, String> = emptyMap()
  @Field var contentId: String? = null
  @Field var multiSession: Boolean = false
  @Field var licenseResponseType: String = "raw"
}

class FastVideoSubtitle : Record {
  @Field var id: String? = null
  @Field var uri: String = ""
  @Field var mimeType: String = "text/vtt"
  @Field var language: String? = null
  @Field var label: String? = null
  @Field var isDefault: Boolean = false
}


class FastVideoMetadata : Record {
  @Field var title: String? = null
  @Field var artist: String? = null
  @Field var albumTitle: String? = null
  @Field var artworkUri: String? = null
}

class FastVideoSource : Record {
  @Field var uri: String = ""
  @Field var type: String = "auto"
  @Field var headers: Map<String, String> = emptyMap()
  @Field var drm: FastVideoDrm? = null
  @Field var subtitles: List<FastVideoSubtitle> = emptyList()
  @Field var isLive: Boolean = false
  @Field var startPositionMs: Double? = null
  @Field var targetLiveOffsetMs: Double? = null
  @Field var customCacheKey: String? = null
  @Field var offlineId: String? = null
  @Field var preloadIndex: Int? = null
  @Field var fallbackUris: List<String> = emptyList()
  @Field var maxRetryAttempts: Int = 2
  @Field var retryBackoffMs: Long = 350
  @Field var latencyMode: String = "balanced"
  @Field var metadata: FastVideoMetadata? = null
  @Field var mediaSession: Boolean = false
}

File: ios/FastVideoRecords.swift
import ExpoModulesCore

internal struct FastVideoDrm: Record {
  @Field var type: String = ""
  @Field var licenseUrl: String = ""
  @Field var certificateUrl: String? = nil
  @Field var headers: [String: String] = [:]
  @Field var contentId: String? = nil
  @Field var multiSession: Bool = false
  @Field var licenseResponseType: String = "raw"
}

internal struct FastVideoSubtitle: Record {
  @Field var id: String? = nil
  @Field var uri: String = ""
  @Field var mimeType: String = "text/vtt"
  @Field var language: String? = nil
  @Field var label: String? = nil
  @Field var isDefault: Bool = false
}


internal struct FastVideoMetadata: Record {
  @Field var title: String? = nil
  @Field var artist: String? = nil
  @Field var albumTitle: String? = nil
  @Field var artworkUri: String? = nil
}

internal struct FastVideoSource: Record {
  @Field var uri: String = ""
  @Field var type: String = "auto"
  @Field var headers: [String: String] = [:]
  @Field var drm: FastVideoDrm? = nil
  @Field var subtitles: [FastVideoSubtitle] = []
  @Field var isLive: Bool = false
  @Field var startPositionMs: Double? = nil
  @Field var targetLiveOffsetMs: Double? = nil
  @Field var customCacheKey: String? = nil
  @Field var offlineId: String? = nil
  @Field var preloadIndex: Int? = nil
  @Field var fallbackUris: [String] = []
  @Field var maxRetryAttempts: Int = 2
  @Field var retryBackoffMs: Double = 350
  @Field var latencyMode: String = "balanced"
  @Field var metadata: FastVideoMetadata? = nil
  @Field var mediaSession: Bool = false
}
